storage/store.go

```diff
@@ -621,13 +611,31 @@ func (s *Store) startGossip() {
 	})
 }
 
-// maybeGossipFirstRange checks whether the store has a replia of the first
-// range and if so, reminds it to gossip the first range descriptor and
-// sentinel gossip.
+// maybeGossipFirstRange checks whether the store has a replica of the
+// first range and if so instructs it to gossip the cluster ID,
+// sentinel gossip and first range descriptor. This is done in a retry
+// loop in the event that the returned error indicates the state of
+// the leader lease is not known. This can happen on lease command
+// timeouts. The retry loop makes sure we try hard to keep asking for
+// the lease instead of waiting for the next clusterIDGossipInterval
+// to transpire.
 func (s *Store) maybeGossipFirstRange() error {
-	rng := s.LookupReplica(roachpb.RKeyMin, nil)
-	if rng != nil {
-		return rng.maybeGossipFirstRange()
+	retryOptions := retry.Options{
+		InitialBackoff: 100 * time.Millisecond, // first backoff at 100ms
+		MaxBackoff:     1 * time.Second,        // max backoff is 1s
+		Multiplier:     2,                      // doubles
+		Closer:         s.stopper.ShouldStop(), // stop no matter what on stopper
+	}
+	for loop := retry.Start(retryOptions); loop.Next(); {
+		rng := s.LookupReplica(roachpb.RKeyMin, nil)
+		if rng != nil {
+			err := rng.maybeGossipFirstRange()
+			if nlErr, ok := err.(*roachpb.NotLeaderError); !ok || nlErr.Leader != nil {
```

Comment:
It's true, but I still wanted to preserve that piece of logic against potential future changes upstream of this call site.